larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.h

```diff
@@ -0,0 +1,177 @@
+/**
+ *  @file   larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewThreeDKinkTool.h
+ *
+ *  @brief  Header file for the two view three d kink tool class.
+ *
+ *  $Log: $
+ */
+#ifndef TWO_VIEW_THREE_D_KINK_TOOL_H
+#define TWO_VIEW_THREE_D_KINK_TOOL_H 1
+
+#include "larpandoracontent/LArObjects/LArPointingCluster.h"
+
+#include "larpandoracontent/LArThreeDReco/LArTwoViewMatching/TwoViewTransverseTracksAlgorithm.h"
+
+namespace lar_content
+{
+
+/**
+ *  @brief  TwoViewThreeDKinkTool class
+ */
+class TwoViewThreeDKinkTool : public TransverseMatrixTool
+{
+public:
+    /**
+     *  @brief  Constructor
+     *
+     *  @param  nCommonClusters the number of common clusters to select
+     */
+    TwoViewThreeDKinkTool();
+
+    /**
+     *  @brief  Destructor
+     */
+    virtual ~TwoViewThreeDKinkTool();
+
+    bool Run(TwoViewTransverseTracksAlgorithm *const pAlgorithm, MatrixType &overlapMatrix);
+
+private:
+    /**
+     *  @brief  Modification class
+     */
+    class Modification
```

Comment:
Minor point, but I would make this a struct, which has its members public by default and probably fits the intent a little better.

Reply:
I assume it would be the same for the Particle class. I am fine with it, although I think sub-classes are used in this way elsewhere in Pandora. Is this something we plan to implement in all similar cases in the future?

Reply:
I don't have a strong view on this (I just think it conveys the intent more clearly), where there are new classes that are just acting as containers for variables that will be publicly accessible, I would make those structs and continue to do so in the future, but I wouldn't go back and replace historical classes of this form with structs.

Reply:
Hi @MariaBrigida, we’ve previously stuck to classes across the board, rather than using structs in our codebase. Doesn’t hurt to have to make it explicit that the members are public and occasionally helps avoid confusion. Take the point that @AndyChappell is making (about intent and suitability), but counter that it’s better to just remain consistent here. Thanks.

Reply:
Purely because this thread is about struct vs classes, the c++ guidelines suggest ways to use struct vs class for implying intent: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-org

Not that this should trump consistency across the codebase...